src/job.ts

```diff
@@ -1086,6 +1089,8 @@ export class Job {
             aliases.push(serviceAlias);
         }
 
+        const uniqueAlias = aliases[aliases.length - 1];
```

Comment:
if you have two services using the same image in the job, the health check will just be checking either one. in those cases it's likely that there are unique alias definitions for the services which we can try to use here.

Reply:
Hmm... I wonder how gitlab.com actually pulls this one of.